airflow/providers/redis/provider.yaml

```diff
@@ -39,7 +39,7 @@ dependencies:
   # as well as unquoting URLS with `urllib.parse.unquote`:
   # https://github.com/redis/redis-py/blob/master/CHANGES
   # TODO: upgrade to support redis package >=4
-  - redis~=3.2
+  - redis~=4.5.5
```

Comment:
Can 3.x still be supported? Also `~=4.5.5` limits the version range to only `>=4.5.5,<4.6.0`, which is likely not the intention?

Reply:
How does `redis>=3.2,<4.5.5` sound to you ?

Reply:
do we wish to have an upper bound? can we not keep just `>=3.2`
I do not have enough context on this. Do we know if there was there a reason that we were not able to support redis >= 3.3.0 and now we're able to do so?  Do we have enough tests in our provider to test the `mixins` and unquoting URLS with `urllib.parse.unquote` as mentioned in the comment?

Reply:
I think the existing test suites isn't enough to cover the concerns mentioned in the comment. I'll try to enrich that in this PR

Reply:
After some digging, I can confirm that the use of redis 4 wouldn't break the existing code base and should be safe to upgrade. The fact that it passes the celery integration test is a hint that it should be ok. 

For the concerns mentioned in the comment: 
- **mixins in redis commands** : This has no impact to airflow since `RedisHook` didn't directly use `redis.client`. If we are using `redis.client`, we'd need to switch to `redis.command` 
- **unquoting URLS with `urllib.parse.unquote`** : This has no impact to airflow as well since we are not using the `from_url` function 

I've just revised the doc and added a example dag to illustrate the use of the existing 3 redis Operator / sensor. I ran pytest in breeze environment & can confirm that the DAG is good to go